### DISCUSSION TITLE
Touch keyboard on Windows for Avalonia 11.0.0-preview4

### DISCUSSION THREAD

**Discussion by mgnslndh** [AvaloniaUI/Avalonia#9974] [Q&A] [answered]
Does Avalonia support the built in touch keyboard when running on Windows? It does not work for me out of the box.

![image](https://user-images.githubusercontent.com/1139118/212348315-de1680a1-98a3-4082-b48f-64beff2626d4.png)

**Comment by maxkatz6** [comment]
It should work just fine with 11.0 previews.
What version are you using?

**Reply by mgnslndh** [reply]
Touching a `TextBox` does not bring up the keyboard. Touching "native" Windows text input like Notepad or similar will bring up the keyboard.

Do I need to use any attribute in XAML or similar to enable it?

- Avalonia
  - Version 11.0.preview4
- Windows
  - Edition: Windows 10 Pro
  - Version: 21H2
  - OS Build: 19044.2364

**Reply by mgnslndh** [reply]
I have tested this with the same Dell S2240T touch screen (external) on Windows 11 22H2 22621.963 with the same results

**Reply by mgnslndh** [reply]
I've added some trace messages to the source so that I can track what happens but I still can't figure out why it is not showing correctly. But I have seen the keyboard pop up during debugging and sometimes after I press outside of the Avalonia window:

![TouchKeyboardNotShowingCorrectly3](https://user-images.githubusercontent.com/1139118/213474703-f39ec480-311d-41b8-ae92-9b4a73acd99c.gif)

@maxkatz6 any ideas of how to troubleshoot this?

**Reply by maxkatz6** [reply]
Just checked on the touch laptop, and it indeed seems to not be supported on windows. But is supported on mobile and wasm.
Can you create an issue from this discussion?

**Reply by mgnslndh** [reply]
Sure! I created #10136

**Comment by mgnslndh** [comment, marked as answer]
Concering Avalonia [11.0.0-preview4](https://www.nuget.org/packages/Avalonia/11.0.0-preview4) @maxkatz6:
> Just checked on the touch laptop, and it indeed seems to not be supported on windows. But is supported on mobile and wasm. Can you create an issue from this discussion?
